File: crypto-arbitrage-platform/web_frontend/src/api/strategy.ts
import axios from 'axios'

const api = axios.create({
  baseURL: '/api/v1'
})

export interface Strategy {
  id: string
  name: string
  strategy_type: string
  enabled: boolean
  params: Record<string, unknown>
}

export interface StrategyListResponse {
  strategies: Strategy[]
  total: number
}

export interface StrategyStatusResponse {
  strategy_id: string
  status: string
  message?: string
}

export const strategyApi = {
  list: () => api.get<StrategyListResponse>('/strategies'),

  get: (id: string) => api.get<Strategy>(`/strategies/${id}`),

  start: (id: string) => api.post<StrategyStatusResponse>(`/strategies/${id}/start`),

  stop: (id: string) => api.post<StrategyStatusResponse>(`/strategies/${id}/stop`),

  update: (id: string, params: Record<string, unknown>) =>
    api.put<StrategyStatusResponse>(`/strategies/${id}`, { params })
}

File: crypto-arbitrage-platform/web_backend/src/models/strategy.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct StrategyResponse {
    pub id: String,
    pub name: String,
    pub strategy_type: String,
    pub enabled: bool,
    pub params: serde_json::Value,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct StrategyStartRequest {
    pub strategy_id: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct StrategyStopRequest {
    pub strategy_id: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct StrategyStatusResponse {
    pub strategy_id: String,
    pub status: String,
    pub message: Option<String>,
}
